shared-plugins/wpcom-elasticsearch/wpcom-elasticsearch.php

```diff
@@ -0,0 +1,821 @@
+
+					break;
+
+				case 'post_type':
+					if ( $query->get( 'post_type' ) && 'any' != $query->get( 'post_type' ) ) {
+						$post_types_via_user = $query->get( 'post_type' );
+					}
+					elseif ( ! empty( $_GET['post_type'] ) ) {
+						$post_types_via_user = explode( ',', $_GET['post_type'] );
+					} else {
+						$post_types_via_user = false;
+					}
+
+					$post_types = array();
+
+					// Validate post types, making sure they exist and are public
+					if ( $post_types_via_user ) {
+						foreach ( (array) $post_types_via_user as $post_type_via_user ) {
+							$post_type_object = get_post_type_object( $post_type_via_user );
+
+							if ( ! $post_type_object || $post_type_object->exclude_from_search )
+								continue;
+
+							$post_types[] = $post_type_via_user;
+						}
+					}
+
+					// Default to all non-excluded from search post types
+					if ( empty( $post_types ) )
+						$post_types = array_values( get_post_types( array( 'exclude_from_search' => false ) ) );
+
+					$es_wp_query_args['post_type'] = $post_types;
+
+					break;
+			}
+		}
+
+		// Date
+		if ( $query->get( 'year' ) ) {
+			if ( $query->get( 'monthnum' ) ) {
+				// Padding
+				$date_monthnum = sprintf( '%02d', $query->get( 'monthnum' ) );
+
+				if ( $query->get( 'day' ) ) {
+					// Padding
+					$date_day = sprintf( '%02d', $query->get( 'day' ) );
+
+					$date_start = $query->get( 'year' ) . '-' . $date_monthnum . '-' . $date_day . ' 00:00:00';
+					$date_end   = $query->get( 'year' ) . '-' . $date_monthnum . '-' . $date_day . ' 23:59:59';
+				} else {
+					$days_in_month = date( 't', mktime( 0, 0, 0, $query->get( 'monthnum' ), 14, $query->get( 'year' ) ) ); // 14 = middle of the month so no chance of DST issues
+
+					$date_start = $query->get( 'year' ) . '-' . $date_monthnum . '-01 00:00:00';
+					$date_end   = $query->get( 'year' ) . '-' . $date_monthnum . '-' . $days_in_month . ' 23:59:59';
+				}
+			} else {
+				$date_start = $query->get( 'year' ) . '-01-01 00:00:00';
+				$date_end   = $query->get( 'year' ) . '-12-31 23:59:59';
+			}
+
+			$es_wp_query_args['date_range'] = array( 'field' => 'date', 'gte' => $date_start, 'lte' => $date_end );
+		}
+
+		// Facets
+		if ( ! empty( $this->facets ) ) {
+			$es_wp_query_args['facets'] = $this->facets;
+		}
+
+		// You can use this filter to modify the search query parameters, such as controlling the post_type.
+		// These arguments are in the format for wpcom_search_api_wp_to_es_args(), i.e. WP-style.
+		$es_wp_query_args = apply_filters( 'wpcom_elasticsearch_wp_query_args', $es_wp_query_args, $query );
+
+		// Convert the WP-style args into ES args
+		$es_query_args = wpcom_search_api_wp_to_es_args( $es_wp_query_args );
+
+		$es_query_args['fields'] = array(
+			'post_id',
+			'blog_id'
+		);
+
+		// Ask for a search suggestion (to catch typos)
+		// See http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-suggesters-phrase.html for options
+		$suggest_beta_blogs = array(
```

Comment:
We can probably delete this or enable the suggest feature for all sites on Go? cc @nickdaugherty